app/store/database.js

```diff
@@ -18,59 +29,49 @@ Database.prototype.updateForm = function(form, callback) {
         type:form.type,
         status:'Draft',
         data:JSON.stringify(form.data)}, 
-        (rows, error)=> {
+        this.sendError(callback, (rows)=> {
             callback(rows[0].last_value);
-        }
+        })
     );
 };
 
 Database.prototype.myCases = function(login, callback) {
-    this.forms.selectByLogin(login, function(rows, error) {
-        if (error) {
-            callback({error:error});
-        }   
-        else {     
-            callback(rows.map(function(row) {
-                let modified = row.modified;
-                modified = JSON.stringify(modified).toString();
-                modified = modified.substring(1, modified.lastIndexOf('.'))+'Z';
-                return {
-                    id: row.id,
-                    type: row.type,
-                    status: row.status,
-                    modified: modified,
-                    data: JSON.parse(row.data)
-                };
-            }));
-        }
-    });
+    this.forms.selectByLogin(login, this.sendError(callback, (rows)=> {
+        callback(rows.map(function(row) {
+            let modified = row.modified;
+            modified = JSON.stringify(modified).toString();
+            modified = modified.substring(1, modified.lastIndexOf('.'))+'Z';
+            return {
+                id: row.id,
+                type: row.type,
+                status: row.status,
+                modified: modified,
+                data: JSON.parse(row.data)
+            };
+        }));
+    }));
 };
 Database.prototype.savePerson = function(person, callback) {
-    this.persons.findByLogin(person.login, (rows, error)=> {
-        if (error) {
-            callback({error:error});
-        } 
+    this.persons.findByLogin(person.login, this.sendError(callback, (rows)=> {
+        if (rows.length ==0) {
+            this.persons.create(person, this.sendError(callback, (rows)=>{
```

Comment:
a thought: calling this function sendError seems to imply it will always have an error to send.  I would like your consider a name like "checkForError" or 'sendDataOrError' or ...

Reply:
I was struggling with finding a correct name.
checkForError is not enough because it checks and send if any error
sendDataOrError is not correct because it does not send data if no error
...
Perhaps a better way is to reproduce the pattern used in responses.js

Reply:
For example:
`
this.persons.create(person, (rows, error)=>{ 
  this.ifNo(error, callback).then(()=>{ callback(rows[0].last_value); });
});
`
Do you like that better?

Reply:
What about this?

    this.persons.create(person, ifError(callback).otherwise((rows)=>{ 
        callback(rows[0].last_value); 
    }));